appserver/monitoring-console/webapp/src/main/java/fish/payara/monitoring/web/MonitoringConsoleResource.java

```diff
@@ -77,10 +78,19 @@ private static RequestTraceStoreInterface getRequestTracingStore() {
         return Globals.getDefaultBaseServiceLocator().getService( RequestTracingService.class).getRequestTraceStore();
     }
 
+    private static Series seriesOrNull(String series) {
+        try {
+            return new Series(series);
+        } catch (RuntimeException e) {
+            return null;
```

Comment:
Better idea - I opened the Series constructor, you can remove the whole try/catch, because it is not required, The safe code can look like this:

1) Change the constructor of the Series class to provide better informations about possible error
```
        Objects.requireNonNull(key, "key");
        final String[] parts = key.split(SPLIT_PATTERN);
        // last part is the metric
        final int countOfTags = parts.length - 1;
        this.tags = new String[countOfTags];
        this.values = new String[countOfTags];
        this.metric = parts[parts.length - 1].intern();
        for (int i = 0; i < countOfTags; i++) {
            int eqIndex = parts[i].indexOf(TAG_ASSIGN);
            if (eqIndex < 0) {
                throw new IllegalArgumentException("The key does not have a proper syntax: " + key);
            }
            tags[i] = parts[i].substring(0, eqIndex).intern();
            values[i] = parts[i].substring(eqIndex + 1).intern();
        }

```

2) Log errors at least on finest level, but do not swallow stack traces:
```
    private static Series seriesOrNull(String series) {
        try {
            return new Series(series);
        } catch (RuntimeException e) {
            LOG.log(Level.WARNING, "Could not instantiate the Series class!", e);
            return null;
        }
    }
```
Side note - are those intern() calls required?

Javadoc with some example of the key variable syntax would be nice.

Reply:
I made sure `Series` always throws a `IllegalArgumentException` with a text indicating what is wrong and documented the format and exception for the constructor. I added tests that check illegal inputs and added logging and further handling of a potential exception to all usages of `new Series` (apart from usages within tests).